ligo-webide-frontend/ligo-ide/.eslintrc.js

```diff
@@ -0,0 +1,192 @@
+module.exports = {
```

Comment:
If imported from a another licensed project we should put a note at the top about where it's from

Reply:
I used `edna` file and modified it (a lot), but, hm, it is like common thing, I mean it is just config, do we really need to do it?

Reply:
AFAIK some files are not "copyrightable", maybe it's one of such files. But anyway, we have a policy about tracking licensing information on per-file basis, so ideally it should be done anyway and we should have a header here. This way licensing information is supposed to be not lost when you copy files. So if you copied it from https://github.com/serokell/edna/blob/master/frontend/.eslintrc.js, it should have the following header:
```
// SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io>
//
// SPDX-License-Identifier: AGPL-3.0-or-later
```

AFAIU we use `CC0-1.0` for files that are likely not "copyrightable". If you think it's such a file, I think we can write
```
// SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io>
//
// SPDX-License-Identifier: CC0-1.0
```

It's probably ok to change the license of files "owned" by us on the fly.

AFAIU neither `AGPL-3.0-or-later` nor `CC0-1.0` should cause any troubles. AGPL imposes some restrictions, but this particular file is not used in the build process, AFAIU it's used only for linting, so whatever gets deployed can be produced with `.eslintrc.js`. And having `SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io>` seems fine, because originally this file was written by us in the codebase owned by us.

Reply:
>originally this file was written by us in the codebase owned by us

Where is that edge when we should add license information? For example I also copied `.prettierrc` but it has only 5 lines of meaningful code and even more I think a lot of `.prettierrc` configs in the world has the same 5 lines of code.

>And having SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io> seems fine

I should add only this line or also `SPDX-License-Identifier: CC0-1.0`?

Reply:
The idea is to add it to all files, so that `reuse lint` passes. But yes, there should be some "edge" between "copyrightable" and "non-copyrightable". AFAIU, the idea is that for "non-copyrightable" files you can always pretend that you wrote them from scratch, because there is no inellectual property in them. So you can write any license for them. And we use `CC0-1.0` for such files because AFAIU it means "do whatever you want".

I don't know whether this `.eslintrc.js` should be considered copyrightable to be honest. But in this case it probably doesn't really matter, because even if it's copyrightable, this copyright belongs to us, so we can do whatever we want with this file.

I think for non-copyrightable files it's not very useful to provide copyright information, but for example specifying `CC0-1.0` or another unlicense-like license you make it explicitly that you don't care about licensing for that file. Another reason to do it is to please automated tooling (`reuse lint`).

> I should add only this line or also `SPDX-License-Identifier: CC0-1.0`?

The whole header. AFAIU it's the format expected by `reuse lint`. And I think we want `reuse lint` to pass eventually.